### PR TITLE
Docs: rewrote allowArrowFunction option /w declaration (fixes #12900)

### LINKED ISSUE #12900: Incorrect documentation for func-style
**Tell us about your environment**

* **ESLint Version:**
* **Node Version:**
* **npm Version:**

**What parser (default, Babel-ESLint, etc.) are you using?**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js

```

```bash

```

**What did you expect to happen?**

The correct code example for the configuration `/*eslint func-style: ["error", "expression"]*/` shouldn't have an arrow function, as they're disallowed by default.

https://eslint.org/docs/rules/func-style#expression

**What actually happened? Please include the actual, raw output from ESLint.**

```
/*eslint func-style: ["error", "expression"]*/

var foo = function() {
    // ...
};

var foo = () => {};
```

**Are you willing to submit a pull request to fix this bug?**

Maybe.

### PR DESCRIPTION
**Pull request by Caryyon**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).
- [x] The team has reached consensus on the changes proposed in this pull request. If not, I understand that the evaluation process will begin with this pull request and won't be merged until the team has reached consensus.

#### What is the purpose of this pull request? (put an "X" next to an item)

[ X ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)
I took a pass at re-writing the allowArrowFunciton + declaration options to make the easier to understand.

#### Is there anything you'd like reviewers to focus on?
Just if the re-write helps get the point across.

### CODE REVIEW

**Comment by Caryyon** [comment]
is there a lag in the signing of the CLA? I signed it with my github account right before making the PR.

**Comment by nzakas** [comment]
Can you double-check that the email address of your GitHub account is the same one you used for the commit in this PR? People sometimes run into this when they commit using a work computer (with work email co figured) instead of their personal one.

File: docs/rules/func-style.md [hunk 1]
```diff
@@ -55,7 +55,7 @@ This rule has a string option:
 
 This rule has an object option for an exception:
 
-* `"allowArrowFunctions": true` (default `false`) allows the use of arrow functions (honoured only when using `declaration`)
+* When using the `"allowArrowFunctions": true` (default `false`) option with `declaration` option, arrow functions are allowed
```
Comment (kaicataldo) [inline docs/rules/func-style.md]: This is inconsistent with the options on line 53 and 54. Thoughts on something along these lines?
```suggestion
* `"allowArrowFunctions": true` (default `false`) When used with the `declaration` option, arrow functions are allowed
```

**Comment by Caryyon** [comment]
@nzakas i've got the proper check but how do i remove the old email from the check?

**Comment by kaicataldo** [comment]
We use the commit author for this check. One of the commits in this PR must have the old email address, so finding it and changing it to the email in the CLA should fix the check.

**Pull request closed without merging**